### ISSUE TITLE
Is there a plan to change levels shown in vr3 panel?

### ISSUE THREAD

**Issue by wangzhaohe** [leo-editor/leo-editor#3274]
@tbpassin All the outline nodes shown in vr3 panel have the same level H2. Is there a plan to change levels shown in vr3 panel?

<img width="280" alt="image" src="https://user-images.githubusercontent.com/2628013/230804399-1280c20b-6cc5-442e-9526-6034fceb7551.png">

<img width="361" alt="image" src="https://user-images.githubusercontent.com/2628013/230804478-6ab6aa6e-1199-4f21-b0db-37f224a66ed5.png">

**Comment by tbpassin** [comment]
At this time I do not have a plan to use different headline levels.  I understand why one would want them.  I have also had a great deal of difficulty with @rst trees and heading levels, especially when I want to add my own section headings within a node.  ReStructuredText is very sensitive to inconsistent heading levels.  And if you view a subtree of your document, the right levels would be different than if you viewed the entire subtree.  So it can be tricky.

I think that you mostly use asciidoc, don't you?  Asciidoc may be less demanding about getting the levels to nest properly, and what delineators to use for the various headings.  If so, I might consider making nested headings work for asciidoc first, and later follow up with MD and maybe ReST if I can figure out a good way to do so. 

Would this approach work for you?  I won't estimate how soon this could happen since I'm working on a number of things at the moment.

**Comment by wangzhaohe** [comment]
Yes, I agree with you. The similar situation with markdown file. So I never use headline in node body when using markdown format. The main reason is the levels will change after read back .md file.

**Comment by wangzhaohe** [comment]
Yes, asciidoctor has a discrete headings, it will deal with the nest headline in body well.

https://docs.asciidoctor.org/asciidoc/latest/blocks/discrete-headings/

**Comment by wangzhaohe** [comment]
> Would this approach work for you? I won't estimate how soon this could happen since I'm working on a number of things at the moment.

@tbpassin I ever read your source code of creating vr3 levels. I will try to learn it again.

**Comment by tbpassin** [comment]
When the code captures the headline (p.h), it should also capture the level (p.level()) and use that to set the headline depth.

**Comment by tbpassin** [comment]
There is another matter that needs to be settled about indenting headings. Should their body text be shown indented too?  When the nesting level is large, this can lead to text content that is hard to read. Leo's at-files avoid this by indenting the `@others` directive but not the displayed body text.  VR3 does not understand `@others`, for various reasons.
